### ISSUE TITLE
Animation does not work for --precision=x where x<4

### ISSUE THREAD

**Issue by pstricks-fans** [mgieseki/dvisvgm#136]
See my repo: https://github.com/pstricks-fans/dvisvgm-precision-issue.

`animator.bat` contains command to invoke `dvisvgm` with `--precision=x`. The animated SVG does not work for  `x<4`. I love `--precision=1` because it makes the animated SVG size small. 

To test it, just run `invoker.bat` without any parameters.

I hope you can fix it someday.

**Comment by mgieseki** [comment]
Sorry, your scripts don't work on my Windows system. I get several errors, including
```
LaTeX Warning: Reference `anim@abspage0' on page 1 undefined on input line 5.

! Package animate Error: Missing frame number in \animategraphics command.

See the animate package documentation for explanation.
Type  H <return>  for immediate help.
 ...

l.5 ...dafter[\Options]{\FrameRate}{\FileName}{}{}
```
Please provide a single self-contained `.tex` file that doesn't need any preprocessing.

**Comment by pstricks-fans** [comment]
You still need `-shell-escape` to compile the following with `pdflatex`.

```

% compile it with pdflatex -shell-escape
\documentclass{article}
\usepackage{filecontents}

 %frames.tex
\begin{filecontents*}{frames.tex}
\documentclass[pstricks,border=12pt]{standalone}
\usepackage{pst-plot}
\begin{document}
\multido{\i=10+10}{36}
{
	\begin{pspicture}[showgrid=false](-2,-2)(2,2)
        \psline(2;\i)      
	\end{pspicture}
}
\end{document}
\end{filecontents*}


% animated.tex
\begin{filecontents*}{animated.tex}
\documentclass[border=12pt]{standalone}
\usepackage[dvisvgm]{animate}
\usepackage[dvisvgm]{graphicx}
\begin{document}
\animategraphics[controls,loop,autoplay,scale=1]{10}{frames}{}{}
\end{document}
\end{filecontents*}



\begin{document}


\immediate\write18{latex frames}
\immediate\write18{dvips frames}
\immediate\write18{ps2pdf frames.ps}



% optionally remove temporary files
\makeatletter
\@for\x:={tex,dvi,ps,log,aux}\do{\immediate\write18{cmd /c del frames.\x}}
\makeatother


\immediate\write18{latex animated}


% optionally make sure the existing animated.svg has been deleted.
\immediate\write18{cmd /c del animated.svg}

% there are some peculiarities for --precision=x where x<6.
\immediate\write18{dvisvgm --bbox=papersize --font-format=woff --zoom=-1 --precision=5 animated}

% optionally remove temporary files
\makeatletter
\@for\x:={tex,dvi,ps,log,aux}\do{\immediate\write18{cmd /c del animated.\x}}
\makeatother

Please view the generated animated.svg! 


\end{document}
```


I will show the peculiarity for `--precision=5` as follows.

![Missing-Next-Button](https://user-images.githubusercontent.com/50270635/85423022-69a46e00-b5a0-11ea-9f4b-4ef8fac5c1f0.png)

**Comment by mgieseki** [comment]
Sorry, I'm still getting the same error as above. Do you use an outdated version of the `animate` package?

**Comment by pstricks-fans** [comment]
I installed TeXLive about 1 month ago. The documentation of `animate` is as follows.

![Untitled](https://user-images.githubusercontent.com/50270635/85425450-652d8480-b5a3-11ea-937a-c9d2261c2915.png)

I think it is not an outdated version.

**Comment by agrahn** [comment]
It boils down to the following: With `--precision=1`, `dvisvgm` does not evaluate or only partially evaluates coordinate expressions `{?(...)}` such as  in `translate({?(-(x))},{?(-(y))})` contained in the DVI.

A minimal example:

````latex
\documentclass{standalone}

\usepackage[dvisvgm]{animate}
\usepackage[dvisvgm]{graphicx}

\begin{document}
\animategraphics[controls,type=pdf]{12}{example-image}{}{}
\end{document}
````
Typeset with `latex` and once with `dvisvgm --exact --font-format=woff2 --precision=1`, a second time with default precision `dvisvgm --exact --font-format=woff2`. The former conversion leaves some `{?(...)}` expressions untouched.

**Comment by mgieseki** [comment]
Thanks for the further hints. I'm still struggling to compile the example file and therefore wasn't able to reproduce the issue with it. 
The MWE did the job and I can confirm it's actually a bug in the evaluation of `{?(...)}`. I've fixed it locally and will commit the patch after some more testing.

**Comment by pstricks-fans** [comment]
If I want to update my installed dvisvgm binary, do I have to wait until TeXLive 2021? Or will you provide the updated binary to let us download?

**Comment by mgieseki** [comment]
TeX Live provides frequent updates only for non-binary packages. Utilities that need rebuilds for all target platforms, like dvisvgm, are only updated with every new TL release. So, unfortunately, you have to wait for TL 2021 or build the latest dvisvgm yourself. TL for Windows is an exception, though, because Akira Kakuto regularly updates the Windows binaries: http://w32tex.org. 
Currently, I'm too busy to release a new version of dvisvgm. Maybe you can ask Akira if he can apply the latest patch to his build of dvisvgm 2.9.1.

**Comment by pstricks-fans** [comment]
Has this issue been fixed?

**Comment by mgieseki** [comment]
Yes, it should have been fixed since version 2.10.
